### PR TITLE
Updated autorest.md with swagger sdk

### PR DESCRIPTION
**Pull request by ankur741999**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by live1206** [comment]
Please resolve CI issue as below
```
Generated code is not up to date.
    You may need to rebase on the latest main, 
    run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
    run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
    run 'dotnet build /t:GenerateCode' to update the generated code.
    run 'dotnet build /t:GenerateTests' to update the generated test code.
    
To reproduce this error locally, run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory recoveryservices'.
```

**Comment by ankur741999** [comment]
Hi @live1206 : I ran the provided commands. Please review.

**Comment by live1206** [comment]
@ankur741999 There is a breaking change in the newly generated SDK. [ImmutabilityState](https://github.com/Azure/azure-sdk-for-net/blob/fbd50bd3d03e6483a85b3f75ed194ded4c29a1e5/sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/Generated/Models/RecoveryServicesVaultProperties.cs#L80C52-L80C52) was removed in the new code. We need to add it back as partial class in custom code to mitigate the breaking change, please refer to [cosmosdb](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Custom) how it was added as custom code. 
Please feel free to ping me if you have question on this.

**Comment by ankur741999** [comment]
Hi @live1206 : 
As discussed offline we have an upcoming GA dependent on this.
GA timelines: 11 Sept
Expected Release date: 5th Sept

Can we please kindly release this by expected date because of our prior commitments given it will affect GA timelines.

File: sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/Generated/Models/SecuritySettings.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 namespace Azure.ResourceManager.RecoveryServices.Models
 {
     /// <summary> Security Settings of the vault. </summary>
-    internal partial class SecuritySettings
+    public partial class SecuritySettings
```
Comment (live1206) [inline sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/Generated/Models/SecuritySettings.cs:11]: `SecuritySettings` is too general, please rename it to a more specific name, such as prepending RP name to it by adding it to `prepend-rp-prefix` in [autorest.md](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/autorest.md)

File: sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/Generated/Models/SoftDeleteSettings.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.RecoveryServices.Models
+{
+    /// <summary> Soft delete Settings of vault. </summary>
+    public partial class SoftDeleteSettings
```
Comment (live1206) [inline sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/Generated/Models/SoftDeleteSettings.cs:11]: The naming is too general, add it to `prepend-rp-prefix` in [autorest.md](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/autorest.md)

File: sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/Generated/Models/SoftDeleteState.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.RecoveryServices.Models
+{
+    /// <summary> The SoftDeleteState. </summary>
+    public readonly partial struct SoftDeleteState : IEquatable<SoftDeleteState>
```
Comment (live1206) [inline sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/Generated/Models/SoftDeleteState.cs:14]: The naming is too general, add it to `prepend-rp-prefix` in [autorest.md](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/recoveryservices/Azure.ResourceManager.RecoveryServices/src/autorest.md)

**Comment by live1206** [comment]
Please also update the changelog with release notes and expecting release date.

**Review by live1206 — APPROVED** [review]

**Pull request merged**